### PR TITLE
Refactor CommandProcessor injection

### PR DESCRIPTION
**Pull request by lindycoder**
To make it easier to extend switch without having to copy everything

### CODE REVIEW

File: fake_switches/cisco/command_processor/enabled.py [hunk 1]
```diff
@@ -281,8 +283,6 @@ def build_running_interface(port):
             data.append(" ip access-group %s out" % port.access_group_out)
         if port.ip_redirect is False:
             data.append(" no ip redirects")
-        if port.ip_proxy_arp is False:
```
Comment (fbouliane) [inline fake_switches/cisco/command_processor/enabled.py]: why was this removed ?

File: fake_switches/brocade/command_processor/config_virtual_interface_vrrp.py [hunk 2]
```diff
@@ -16,12 +16,9 @@
...
 
 class ConfigVirtualInterfaceVrrpCommandProcessor(ConfigInterfaceCommandProcessor):
-    def __init__(self, switch_configuration, terminal_controller, logger,
-                 piping_processor, port, vrrp):
-        super(ConfigVirtualInterfaceVrrpCommandProcessor, self).__init__(
-            switch_configuration, terminal_controller, logger, piping_processor,
-            port)
-        self.vrrp = vrrp
+    def init(self, switch_configuration, terminal_controller, logger, piping_processor, *args):
+        super(ConfigVirtualInterfaceVrrpCommandProcessor, self).init(switch_configuration, terminal_controller, logger, piping_processor, args[0])
```
Comment (fbouliane) [inline fake_switches/brocade/command_processor/config_virtual_interface_vrrp.py]: what is args[0] here ?
For me it's unclear.

I suggest to take a better contract than *args. My suggestion is **kwargs but not sure if it's the best.
Another solution would be to name this parameter prior passing it to ConfigVirtualInterfaceVrrpCommandProcessor

File: fake_switches/command_processing/base_command_processor.py [hunk 3]
```diff
@@ -16,8 +16,7 @@
 
 
 class BaseCommandProcessor(CommandProcessor):
-
-    def __init__(self, switch_configuration, terminal_controller, logger, piping_processor):
+    def init(self, switch_configuration, terminal_controller, logger, piping_processor):
```
Comment (fbouliane) [inline fake_switches/command_processing/base_command_processor.py]: shouldn't it take *args as a last parameter like child's implementation ?

Comment (lindycoder) [inline fake_switches/command_processing/base_command_processor.py]: yes

File: fake_switches/command_processing/base_command_processor.py [hunk 4]
```diff
@@ -82,8 +81,14 @@ def delegate_to_sub_processor(self, line):
...
-    def move_to(self, new_process_class, *args):
-        self.sub_processor = new_process_class(self.switch_configuration, self.terminal_controller, self.logger, self.piping_processor, *args)
+    def move_to(self, new_processor, *args):
+        new_processor.init(self.switch_configuration,
+                           self.terminal_controller,
+                           self.logger,
+                           self.piping_processor,
+                           *args)
+        self.sub_processor = new_processor
+        self.logger.info("new sub = {}".format(self.sub_processor.__class__.__name__))
```
Comment (lindycoder) [inline fake_switches/command_processing/base_command_processor.py]: yes

File: fake_switches/cisco/command_processor/enabled.py (hunk 1 above)
Comment (lindycoder) [inline fake_switches/cisco/command_processor/enabled.py]: C'est une erreur de merge on dirais

Comment (lindycoder) [inline fake_switches/cisco/command_processor/enabled.py]: nice catch

File: fake_switches/brocade/command_processor/config_virtual_interface_vrrp.py (hunk 2 above)
Comment (lindycoder) [inline fake_switches/brocade/command_processor/config_virtual_interface_vrrp.py]: Actually the inheritance to the interface processor made no sense as the VRRP processor do not share the commands... so i'll change the parent instead and the params assign becomes

        self.port, self.vrrp = args
